### PR TITLE
fix: added naive intersection check for bypassing threshold

### PR DESCRIPTION
**Pull request by ChaoticTempest**
This will naively pop and insert back triples and presigs where the intersection between their participants and the current active participants do not meet the threshold requirement. This is naive and can have bad worst time tradeoffs but better to have it than not for now

### CODE REVIEW

**Review by volovyks — APPROVED** [review]

**Comment by volovyks** [comment]
I have executed the tests one more time, it's throwing `ExecutionError("Exceeded the prepaid gas.")`  in `test_signature_large_stockpile`.

**Comment by ChaoticTempest** [comment]
@volovyks yeah it's probably because the GH actions instance is slow to be running 8 nodes at once

File: node/src/protocol/presignature.rs [hunk 1]
```diff
@@ -246,6 +246,11 @@ impl PresignatureManager {
         self.presignatures.remove(&id)
     }
 
+    pub fn insert_mine(&mut self, presig: Presignature) {
+        self.mine.push_back(presig.id);
```
Comment (ppca) [inline node/src/protocol/presignature.rs:250]: nice, so basically it's deprioritized

**Review by ppca — APPROVED** [review]
LGTM. we could optimize how to pick the triple/presignature that has the largest overlap later.

**Review by volovyks — DISMISSED** [review]

**Review by ppca — APPROVED** [review]

**Pull request merged**